doc/migrations.md

```diff
@@ -5,26 +5,32 @@
 If you are planning to use database migrations, you need to let
 Alembic know the current state of the database.
 
-Presuming you've just started off by running `make load_demo_data`,
+Presuming you've just started off by running `make load_demo_data`
+on the latest master branch commit (do not check out your database schema changes yet),
 tell Alembic that you are on the latest database schema:
 
 ```
 PYTHONPATH=. alembic -x config=config.yaml stamp head
 ```
 
-## Upgrade to latest
+## Generate migration script
 
-Subsequently, when the database schema changes, run the following to
-upgrade:
+To generate a migration script, after having stamped the latest master commit (above),
+ensure that the app is stopped, check out the branch with the relevant DB schema
+changes, and run the following:
 
 ```
-PYTHONPATH=. alembic -x config=config.yaml upgrade head
+PYTHONPATH=. alembic -x config=config.yaml revision --autogenerate -m "Revision description"
 ```
 
-## Generate migration
+Review the resulting migration file under `alembic/versions` at hand of the [documentation](https://alembic.sqlalchemy.org/en/latest/autogenerate.html).
+
+
+## Applying migration scripts
+
+Subsequently, after ensuring that the app is stopped, check out the branch with
+pending migration scripts, and run the following to upgrade:
```

Comment:
Should proably add a note about migration manager, noting that this happens automatically upon firing up the app, and that the file `log/migration.log` shows more details.